File: backend/src/Devsu.Application/Dtos/Transactions/GetTransaction.cs
using Devsu.Application.Dtos.Accounts;

namespace Devsu.Application.Dtos.Transactions;

public class GetTransaction
{
    public string? Type { get; set; }
    public Guid Id { get; set; }
    public decimal CurrentBalance { get; set; }
    public decimal Amount { get; set; }
    public string? Movement { get; set; }
    public Guid? AccountId { get; set; }
    public virtual GetAccount? Account { get; set; }
    public DateTime CreatedAt { get; set; }
    public string? CreatedAtStr => 
        CreatedAt.ToString("dd-MM-yyyy", System.Globalization.CultureInfo.InvariantCulture);

    public bool Status { get; set; }
}

File: front-angular/src/app/transactions/interfaces/get-transaction.ts
import { GetAccount } from "../../accounts/interfaces/get-account";

export interface GetTransaction {
  id: string;
  type: "debito" | "credito";
  amount: number;
  currentBalance: number;
  movement: string;
  accountId: string;
  account?: GetAccount;
  createdAtStr: string;
  status: boolean;
}